File: tp01/exercicio1.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
int tamanhoString(char* string){
    int c = 0;
    while(string[c] != '\0'){
        c++;
    }
    return c;
}

bool palindromo(char* string) //Função que verifica se uma string é um palíndromo
{
    bool ehPalindromo = true;
    int  i = 0;
    while(i < tamanhoString(string)/2 && ehPalindromo) //Loop para comparar os caracteres da string
    {
        if(string[i] != string[tamanhoString(string) - 1 - i]) //Compara o caractere da posição i com o caractere da posição correspondente do final
        {
            ehPalindromo = false; //Se os caracteres não forem iguais, ou forem caracteres não letras, não é um palíndromo
        }
        i++;
    }
    return ehPalindromo;
}


int main()
{
    char string[500];
    
    scanf(" %[^\n]", string);
    while(strcmp(string, "FIM") != 0) //Loop que lê strings até o fim do arquivo (determinado pela string FIM)
    {
        if(palindromo(string)) printf("SIM\n"); //Imprime o resultado("SIM" se for palindromo e "NAO" se nao for palindromo)
        else printf("NAO\n");
        scanf(" %[^\n]", string);
    }

    return 0;
}

File: tps/tp01/ex01/Exercicio1.java

class Exercicio1{
    public static void main(String[] args) {
        String string = MyIO.readLine();
        while(!"FIM".equals(string)){//Loop que lê strings até o fim do arquivo (determinado pela string FIM)
            if(ehPalindromo(string)) MyIO.println("SIM"); //Imprime o resultado("SIM" se for palindromo e "NAO" se nao for palindromo)
            else MyIO.println("NAO");
            string = MyIO.readLine();
        }
    }
   
    private static boolean ehPalindromo(String string){//Função que verifica se uma string é um palíndromo
        boolean ehPalindromo = true;
        int  i = 0;
        while(i < string.length()/2 && ehPalindromo) //Loop para comparar os caracteres da string
        {
            if(string.charAt(i) != string.charAt(string.length() -1 -i)) //Compara o caractere da posição i com o caractere da posição correspondente do final
            {
                ehPalindromo = false; //Se os caracteres não forem iguais, ou forem caracteres não letras, não é um palíndromo
            }
            i++;
        }
        return ehPalindromo;
        }
}